lib/kino/data_table.ex

```diff
@@ -0,0 +1,223 @@
+defmodule Kino.DataTable do
+  @moduledoc """
+  A widget for interactively viewing enumerable data.
+
+  The data must be an enumerable of records, where each
+  record is either map, keyword list or tuple. Structs
+  however must be manually converted to maps.
+
+  ## Examples
+
+      data = [
+        %{id: 1, name: "Elixir", website: "https://elixir-lang.org"},
+        %{id: 2, name: "Erlang", website: "https://www.erlang.org"}
+      ]
+
+      Kino.DataTable.start(data)
+
+  The tabular view allows you to quickly preview the data
+  and analyze it thanks to sorting capabilities.
+
+      data = Process.list() |> Enum.map(&Process.info/1)
+
+      Kino.DataTable.start(
+        data,
+        keys: [:registered_name, :initial_call, :reductions, :stack_size]
+      )
+  """
+
+  use GenServer, restart: :temporary
+
+  defstruct [:pid]
+
+  @type t :: %__MODULE__{pid: pid()}
+
+  @typedoc false
+  @type state :: %{
+          parent_monitor_ref: reference(),
+          data: Enum.t(),
+          total_rows: non_neg_integer()
+        }
+
+  @doc """
+  Starts a widget process with enumerable tabular data.
+  """
+  @spec start(Enum.t(), keyword()) :: t()
+  def start(data, opts \\ []) do
+    validate_data!(data)
```

Comment:
I don't want to validate the data upfront because I want to support passing lazy enumerables. If we do a full traversal upfront, then that can be an issue. :)

Reply:
It is best to validate on every page, as we render them.

Reply:
This is the only reasonable place to validate though, before the process is started. Should we assume the data is valid and then let the widget crash silently?

Reply:
Perhaps we can validate if it is a list? And then skip it otherwise? You can add tests with a `MapSet` of maps as counter examples...

Reply:
Sounds good to me!